### PR TITLE
infections for hospital kernel

### PR DESCRIPTION
**Pull request by tozanski**

### CODE REVIEW

File: julia/src/Simulation/Simulation.jl [hunk 1]
```diff
@@ -73,10 +70,10 @@ function simulate!(state::SimState,
   nothing
 end
   
-using FunctionWrappers
+#using FunctionWrappers
```
Comment (eMaerthin) [inline julia/src/Simulation/Simulation.jl:73]: remove if not used

File: julia/src/Simulation/enums.jl [hunk 2]
```diff
@@ -6,4 +6,4 @@
 
 @enum DetectionStatus::UInt8 Undetected UnderObservation TestPending Detected #2 bits
 
-@enum ContactKind::UInt8 NoContact=0 HouseholdContact FriendshipContact SporadicContact ConstantKernelContact OutsideContact # bits
\ No newline at end of file
+@enum ContactKind::UInt8 NoContact=0 HouseholdContact HospitalContact FriendshipContact SporadicContact ConstantKernelContact OutsideContact # bits
```
Comment (eMaerthin) [inline julia/src/Simulation/enums.jl:9]:
```suggestion
@enum ContactKind::UInt8 NoContact=0 HouseholdContact HospitalContact FriendshipContact SporadicContact ConstantKernelContact OutsideContact # 3 bits
``` 
:D

File: julia/src/Simulation/infection_kernels.jl [hunk 3]
```diff
@@ -80,3 +75,40 @@ function enqueue_transmissions!(state::SimState, ::Type{Val{HouseholdContact}},
   end
 end
 
+function enqueue_transmissions!(state::SimState, ::Type{Val{HospitalContact}}, source_id::Integer, params::SimParams)
+  if nothing === params.hospital_kernel_params
+    return
+  end
+  progression = progressionof(params, source_id)
+  
+  @assert progression.severity in SA[Severe, Critical]
```
Comment (eMaerthin) [inline julia/src/Simulation/infection_kernels.jl:84]: what if patient is with Mild symptoms and he is just undetected and stays in hospital?
if you are severe/critical probably the chances are lower to infect other in the hospital, right @onomatopeia ?

File: julia/src/Simulation/simparams.jl [hunk 4]
```diff
@@ -23,13 +23,19 @@ using Distributions
 #  household_ptrs::Vector{Tuple{UInt32,UInt32}}  # (i1,i2) where i1 and i2 are the indices of first and last member of the household
 #  
 #end
+struct HospitalInfectionParams
+  hospital_staff_ids::Vector{UInt32}
+  kernel_constant::Float64
```
Comment (eMaerthin) [inline julia/src/Simulation/simparams.jl:28]: why `kernel_constant` not `kernel_hospital`?

File: julia/src/Simulation/infection_kernels.jl (hunk 3 above)
Comment (tozanski) [reply julia/src/Simulation/infection_kernels.jl:84]: mild do not go to the hospital in the current version

File: julia/src/Simulation/simparams.jl (hunk 4 above)
Comment (tozanski) [reply julia/src/Simulation/simparams.jl:28]: because it is a constant not a hospital ;)

File: julia/src/Simulation/Simulation.jl (hunk 1 above)
Comment (tozanski) [inline julia/src/Simulation/Simulation.jl:73]:
```suggestion
```

File: julia/src/Simulation/infection_kernels.jl (hunk 3 above)
Comment (eMaerthin) [reply julia/src/Simulation/infection_kernels.jl:84]: but they can be in the hospital not due to covid but because of other diseases they have

**Pull request merged**